### ISSUE TITLE
`s` formatting collapses numbers

### ISSUE THREAD

**Issue by domoritz** [vega/datalib#50]
```js
$ dl.range(20).map(dl.format.auto.number([0, 20], 20, 's'))
["0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "10", "10", "10", "10", "20", "20", "20", "20", "20"]

$ dl.range(20).map(dl.format.auto.number([0, 20], 20, ''))
["0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19"]
```

It looks like we don't get enough significant digits in https://github.com/vega/datalib/blob/master/src/format.js#L99

**Comment by domoritz** [comment]
Ahh, we have to provide a complete domain.

```js
$ dl.range(20).map(dl.format.auto.number(dl.range(20), 0, 's'))
["0.0", "1.0", "2.0", "3.0", "4.0", "5.0", "6.0", "7.0", "8.0", "9.0", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19"]
```

**Comment by domoritz** [comment]
Reopening because things are still odd. 

Shouldn't the output be `0, 1, 2, .... 18, 19` rather than `0.0, 1.0, ...`? The d3-format used to do this without any significant digit calculation. 

```js
$ var fmt = d3.format('s')$ [0,1,10,11,100,110,1000,1100,10000,11000,100000,110000,1000000,1100000].map(function(d) { return fmt(d); })
["0", "1", "10", "11", "100", "110", "1k", "1.1k", "10k", "11k", "100k", "110k", "1M", "1.1M"]
$ dl.range(20).map(function(d) { return fmt(d); })
["0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19"]
```

d3-format seems to format with a default number of significant digits rather than just taking the most significant one. https://github.com/d3/d3-format/issues/2 https://github.com/d3/d3-format/issues/3

```js
$ f = require("d3-format")
$ f.format('s')(10)
"10.0000"
```

Correct me if I'm wrong but I think we want a formatter that uses up to k significant digits (calculated from a domain) but does not print unnecessary 0s.

**Comment by domoritz** [comment]
So we may expect https://github.com/vega/datalib/pull/52 or https://github.com/vega/datalib/pull/53

**Comment by jheer** [comment]
Addressed by #54.
